hbase-asyncfs/src/main/java/org/apache/hadoop/hbase/util/RecoverLeaseFSUtils.java

```diff
@@ -127,15 +134,16 @@ private static boolean recoverDFSFileLease(final DistributedFileSystem dfs, fina
             Thread.sleep(conf.getInt("hbase.lease.recovery.pause", 1000));
             if (findIsFileClosedMeth) {
               try {
+                SafeMode fsWithSafeMode = (SafeMode) fs;
                 isFileClosedMeth =
-                  dfs.getClass().getMethod("isFileClosed", new Class[] { Path.class });
+                  fsWithSafeMode.getClass().getMethod("isFileClosed", new Class[] { Path.class });
```

Comment:
This reflection is not needed. LeaseRecoverable has isFileClosed()

Reply:
I should have resolve this concern in the latest patch, but I'm trying to get green build...